src/server/svr_mail.c

```diff
@@ -752,16 +757,21 @@ svr_mailowner_id(char *jid, job *pjob, int mailpoint, int force, char *text)
 	mcpid = fork();
 	if(mcpid == 0) {
 		/* this child will be sendmail with its stdin set to the pipe */
-      if (mfds[0] != 0) {
-            (void)close(0);
-            if (dup(mfds[0]) == -1)
+		if (mfds[0] != 0) {
+			(void)close(0);
+			if (dup(mfds[0]) == -1)
 				exit(1);
-        }
-        (void)close(1);
-        (void)close(2);
-        if (execv(SENDMAIL_CMD, margs) == -1)
+		}
+		(void)close(1);
+		(void)close(2);
+		if (execv(SENDMAIL_CMD, margs) == -1)
 			exit(1);
 	}
+	if (mcpid == -1) {/* Error on fork */
+		log_err(errno, __func__, "fork failed\n");
+		(void)close(mfds[0]);
+		exit(0);
```

Comment:
In the previous check for fork failure, you return out of the function.  Here you exit().  Why the difference?  Do we really want to exit the server on a fork() failure?  I usually try avoiding such measures for daemons.

Reply:
This is the second fork in the row. If we fail the first fork we are still the main process - so the return is necessary, but after the first fork, the process only intends to send the e-mail, which means the exit is suitable after the second fork. 

Now, I see that it should be exit(1) instead of exit(0).

Reply:
Oh, I see.  This fork() is in the child, so exiting out of the child is the right thing to do.  Yeah, I agree that exit(1) is better than 0.

Reply:
Changed to exit(1).